src/diart/models.py

```diff
@@ -1,29 +1,44 @@
 from abc import ABC, abstractmethod
-from typing import Optional, Text, Union, Callable
+from typing import Optional, Text, Union, Callable, TYPE_CHECKING
 
 import torch
 import torch.nn as nn
+from requests import HTTPError
 
 try:
-    import pyannote.audio.pipelines.utils as pyannote_loader
+    from pyannote.audio import Inference, Model
+    from pyannote.audio.pipelines.speaker_verification import (
+        PretrainedSpeakerEmbedding,
+    )
 
     _has_pyannote = True
 except ImportError:
     _has_pyannote = False
 
+if TYPE_CHECKING:
+    from pyannote.audio import Model
+    from pyannote.audio.pipelines.speaker_verification import (
+        PretrainedSpeakerEmbedding,
+    )
+
 
 class PyannoteLoader:
     def __init__(self, model_info, hf_token: Union[Text, bool, None] = True):
         super().__init__()
         self.model_info = model_info
         self.hf_token = hf_token
 
-    def __call__(self) -> nn.Module:
-        return pyannote_loader.get_model(self.model_info, self.hf_token)
+    def __call__(self) -> Union[Model, PretrainedSpeakerEmbedding]:
```

Comment:
```suggestion
    def __call__(self) -> Callable:
```
@juanmc2005 like so?

Reply:
yes!